### PR TITLE
Add Minimal sample

### LINKED ISSUE #937: Sample (& doc update) that uses simpler Asp.net template format
.NET 6 preview 4 is set to contain a new simpler template for web sites. We should have a sample (and docs) that show how to use YARP with the simpler template.

### PR DESCRIPTION
**Pull request by MihaZupan**
Fixes #937

I will add doc updates to this PR, opening as a draft just because I think it's cool we're at < 10 non-empty LOC to have a functioning reverse proxy.

### CODE REVIEW

File: samples/ReverseProxy.Minimal.Sample/Program.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddReverseProxy()
+    .LoadFromConfig(builder.Configuration.GetSection("ReverseProxy"));
+
+var app = builder.Build();
+
+app.MapReverseProxy();
+
+app.Run();
```
Comment (Tratcher) [inline samples/ReverseProxy.Minimal.Sample/Program.cs:13]: Nice
cc: @halter73 @samsp-msft @davidfowl

**Comment by Tratcher** [comment]
> I will add doc updates to this PR

Let's not add this to our docs right now, only a readme in the sample itself explaining this demonstrates new 6.0 features. We can revisit the docs after 1.0 when we officially start supporting 6.0.

File: samples/ReverseProxy.Minimal.Sample/Program.cs (hunk 1 above)
Comment (samsp-msft) [reply samples/ReverseProxy.Minimal.Sample/Program.cs:13]: If we want to make david happy, could we reduce this down even more?

``` C#
using Microsoft.AspNetCore.Builder;
using Microsoft.Extensions.DependencyInjection;

var builder = WebApplication.CreateBuilder(args);

builder.Services.AddReverseProxy()
    .LoadFromConfig(builder.Configuration.GetSection("ReverseProxy"));

builder.Build()
  .MapReverseProxy();
  .Run();
```

**Comment by samsp-msft** [comment]
can we make it clear in the directory name that this is for .NET 6+

File: samples/ReverseProxy.Minimal.Sample/Program.cs (hunk 1 above)
Comment (MihaZupan) [reply samples/ReverseProxy.Minimal.Sample/Program.cs:13]: Not right now since `MapReverseProxy()` returns a different interface

**Comment by MihaZupan** [comment]
> Let's not add this to our docs right now, only a readme in the sample itself explaining this demonstrates new 6.0 features. We can revisit the docs after 1.0 when we officially start supporting 6.0.

Will do. For users running a 6.0 SDK, following the getting started doc may be slightly more confusing since the behavior of `dotnet new web` changed.

**Comment by MihaZupan** [comment]
> can we make it clear in the directory name that this is for .NET 6+

Called that out in the README

File: samples/ReverseProxy.Minimal.Sample/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  app.MapReverseProxy();
+  ```
+- Adding a `ReverseProxy` section to `appsettings.json`
+
+  The [configuration file](appsettings.json) in this sample represents a minimal configuration for a basic YARP application.
+  See the [Config-based YARP Sample configuration] for a more exhaustive list of settings exposed via configuration.
+
+[Config-based YARP Sample]: ../ReverseProxy.Config.Sample
+[Config-based YARP Sample configuration]: ../ReverseProxy.Config.Sample/appsettings.json
+[Minimal Hosting for ASP.NET Core]: https://github.com/dotnet/aspnetcore/issues/30354
```
Comment (MihaZupan) [inline samples/ReverseProxy.Minimal.Sample/README.md:25]: > `https://github.com/dotnet/aspnetcore/issues/30354`

We should replace this with a blog post once available (is there one already?)

File: samples/ReverseProxy.Minimal.Sample/appsettings.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        "Match": {
+          "Path": "{**catch-all}"
+        }
+      }
+    },
+    "Clusters": {
+      "minimumcluster": {
+        "Destinations": {
+          "example.com": {
+            "Address": "https://httpbin.org/"
```
Comment (alnikola) [inline samples/ReverseProxy.Minimal.Sample/appsettings.json:24]: Are we sure we can use this URL here?

**Review by alnikola — APPROVED** [review]
LGTM

File: samples/ReverseProxy.Minimal.Sample/appsettings.json (hunk 3 above)
Comment (MihaZupan) [reply samples/ReverseProxy.Minimal.Sample/appsettings.json:24]: I explicitly avoided using an URL like http://www.example.com as it issues a redirect to the original host and is as such not useful in an example.
httpbin plays nice behind a reverse proxy - a user running the sample can actually see YARP working.

Httpbin is a well recognized service, it's used in many other MS samples

File: samples/ReverseProxy.Minimal.Sample/README.md (hunk 2 above)
Comment (Tratcher) [reply samples/ReverseProxy.Minimal.Sample/README.md:25]: The preview4 post should be up tomorrow for Build.

File: samples/README.md [hunk 4]
```diff
@@ -16,6 +16,7 @@ The following samples are provided:
 |------- | ----- |
 | [Basic Yarp Sample](BasicYarpSample) | A simple sample that shows how to add YARP to the empty ASP.NET sample to create a fully functioning reverse proxy. | 
 | [Configuration](ReverseProxy.Config.Sample) | Shows all the options that are available in the YARP config file |
+| [Minimal](ReverseProxy.Minimal.Sample) | Shows a minimal config-based YARP application using [Minimal Hosting for ASP.NET Core](https://github.com/dotnet/aspnetcore/issues/30354) |
```
Comment (Tratcher) [inline samples/README.md:19]:
```suggestion
| [Minimal](ReverseProxy.Minimal.Sample) | Shows a minimal config-based YARP application using .NET 6's [Minimal Hosting for ASP.NET Core](https://github.com/dotnet/aspnetcore/issues/30354) |
```

File: samples/ReverseProxy.Minimal.Sample/appsettings.json (hunk 3 above)
Comment (Tratcher) [reply samples/ReverseProxy.Minimal.Sample/appsettings.json:24]: Then rename the destination to match 😁.

**Review by Tratcher — APPROVED** [review]
Wait for the blog post to be available.

File: samples/ReverseProxy.Minimal.Sample/appsettings.json (hunk 3 above)
Comment (MihaZupan) [reply samples/ReverseProxy.Minimal.Sample/appsettings.json:24]: Ah, missed that, will do :)

File: samples/ReverseProxy.Minimal.Sample/appsettings.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        "ClusterId": "minimumcluster",
+        "Match": {
+          "Path": "{**catch-all}"
+        }
+      }
+    },
+    "Clusters": {
+      "minimumcluster": {
+        "Destinations": {
+          "example.com": {
```
Comment (MihaZupan) [inline samples/ReverseProxy.Minimal.Sample/appsettings.json:23]:
```suggestion
          "httpbin.org": {
```

File: samples/ReverseProxy.Minimal.Sample/README.md (hunk 2 above)
Comment (Tratcher) [reply samples/ReverseProxy.Minimal.Sample/README.md:25]: https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-4/#introducing-minimal-apis

**Pull request merged**